script/initialField.js

```diff
@@ -0,0 +1,148 @@
+        
+        generateIdOpenCard: function(field) {
+            var card = initialFieldModule.chooseCard();
+            initialFieldModule.generateOpenCard(field, card);
+            
+            return card;
+        },
+        
+        generateOpenCard: function(field, thisCard) {
+            var newCard = document.createElement("div");
+            var newText = document.createElement("span");
+            newCard.id = thisCard.getIdCard();
+            newCard.setAttribute("class", "image");
+            field.appendChild(newCard);
+            newCard.setAttribute("draggable", "true");
+            newCard.setAttribute("ondragstart", "drag(event)");
+            newCard.setAttribute("data-n", thisCard.getCardWeight());
+            newCard.setAttribute("data-s", thisCard.getSuit());
+            newText.innerHTML = thisCard.getIdCard();
+            initialFieldModule.setColorText(thisCard.getColor(), newText);
+            newCard.appendChild(newText);
+        },
+        
+        removeChildren: function(node) {
+            var children = node.childNodes;
+            while(children.length) {
+                node.removeChild(children[0]);
+            }
+        },
+        
+        initiateBotField: function() {
+            var NUMBER_BOT_DIV = 7;
+            var NUMBER_CLOSE_DIV = 6;
+            for (var j = NUMBER_BOT_DIV; j > 0; j--) {
+                var currentField = document.getElementById("div" + (NUMBER_CLOSE_DIV + j));
+                initialFieldModule.removeChildren(currentField);
+                currentField.setAttribute("ondrop", "drop(event)");
+                currentField.setAttribute("ondragover", "allowDrop(event)");
+                for(var i = 0; i < j - 1; i++) {
+                    currentField = initialFieldModule.generateCloseCard(currentField);
+                }
+                initialFieldModule.generateIdOpenCard(currentField);
+            }
+        },
+        
+        initiateTopField: function() {
+            var NUMBER_TOP_DIV = 4;
+            var ID_FIRST_DIV = 3;
+            for (var j = 0; j < NUMBER_TOP_DIV; j++) {
+                var currentField = document.getElementById("div" + (ID_FIRST_DIV + j));
+                initialFieldModule.removeChildren(currentField);
+                currentField.setAttribute("ondrop", "drop(event)");
+                currentField.setAttribute("ondragover", "allowDrop(event)");
+            }
+        },
+        
+        restartGame: function() {
+            initialFieldModule.createCardsArr();
+            initialFieldModule.initiateBotField();
+            initialFieldModule.initiateTopField();
+            initialFieldModule.removeChildren(div1);
+            initialFieldModule.removeChildren(div2);
+            initialFieldModule.generateCloseCard(div1);
+            gameModule.resetAdditionalCards();
+            initialFieldModule.createAdditionalCardsArr();
+            gameModule.resetCounter();
+        }
+    };
+}());
```

Comment:
Nice solution and code. Still it is possible to solve the duplication of card while dragging.
The code well-structured and clear. I hardly can do any complains.